homeassistant/components/cloud/alexa_config.py

```diff
@@ -199,14 +200,12 @@ def _migrate_alexa_entity_settings_v1(self):
             # Don't migrate if there's a YAML config
             return
 
-        for state in self.hass.states.async_all():
-            async_expose_entity(
-                self.hass,
-                CLOUD_ALEXA,
-                state.entity_id,
-                self._should_expose_legacy(state.entity_id),
+        for entity_id in set(
+            chain(
+                (state.entity_id for state in self.hass.states.async_all()),
+                (entity_id for entity_id in self._prefs.alexa_entity_configs),
             )
-        for entity_id in self._prefs.alexa_entity_configs:
+        ):
```

Comment:
Can be further simplified:
```python
        for entity_id in {
            *self.hass.states.async_all(),
            *self._prefs.alexa_entity_configs,
        }:
```

Reply:
That means we unpack the iterators instead of passing the iterators in, I'm not sure that's a beneficial change?
Anyhow, I changed like that, it's maybe a little bit easier to read.

Reply:
It's all being turned into a set so it doesn't really matter.